internal/server/http/dump.go

```diff
@@ -0,0 +1,116 @@
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"strings"
+)
+
+// RequestData struct to hold request data
+type RequestData struct {
+	Method string              `json:"method"`
+	Host   string              `json:"host"`
+	URL    string              `json:"url"`
+	Header map[string][]string `json:"header"`
+	Body   string              `json:"body"`
+}
+
+func GetRequestData(r *http.Request) *RequestData {
+	bodyBytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Error reading request body: %v", err)
+		return nil
+	}
+	r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes)) // Reset the body
+
+	return &RequestData{
+		Method: r.Method,
+		Host:   r.Host,
+		URL:    r.URL.String(),
+		Header: r.Header,
+		Body:   string(bodyBytes),
+	}
+}
+
+func LogRequest(r *RequestData, s *Server) {
+	if !s.verbose {
+		return
+	}
+
+	// rebuild the request
+	req, err := http.NewRequest(r.Method, r.URL, bytes.NewBufferString(r.Body))
+	req.Host = r.Host
+	if err != nil {
+		log.Printf("failed to create request for logging: %+v", err)
+		return
+	}
+	for key, values := range r.Header {
+		for _, value := range values {
+			req.Header.Add(key, value)
+		}
+	}
+
+	dumped, err := httputil.DumpRequest(req, true)
+	if err != nil {
+		log.Printf("failed to dump request: %v", err)
+	}
+	if dumped != nil {
+		log.Println(string(dumped))
+	}
+
+}
+
+func RecordRequest(r *RequestData, s *Server) {
+	if len(s.dumpRequestsPath) < 1 || s.dumpCh == nil {
+		return
+	}
+	s.dumpCh <- r
+}
+
+// Goroutine function to write requests to a JSON file
+func RequestWriter(filePath string, requestChan <-chan *RequestData) {
+	file, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		log.Fatalf("Failed to open file: %+v", err)
```

Comment:
may be add return, since the `file` would be nil object and it might panic.

Reply:
log.Fatalf is equivalent to Printf() followed by a call to os.Exit(1). So adding a return here would not help unless I am missing something.

Reply:
Do you not want it to exit the app in this case?

Reply:
Got it, sorry for misleading...